app/submitter/convert_payload_0_0_1.py

```diff
@@ -73,48 +82,55 @@ def convert_answers_to_payload_0_0_1(
     return data
 
 
-def _format_downstream_answer(answer_type, answer_value, answer_data):
-    if answer_type == "Date":
-        return (
-            datetime.strptime(answer_value, "%Y-%m-%d")
-            .replace(tzinfo=timezone.utc)
-            .strftime("%d/%m/%Y")
-        )
+def _format_downstream_answer(
+    answer_type: str, answer_value: AnswerValueTypes, answer_data: str
+) -> str:
+    if isinstance(answer_value, str):
+        if answer_type == "Date":
+            return (
+                datetime.strptime(answer_value, "%Y-%m-%d")
+                .replace(tzinfo=timezone.utc)
+                .strftime("%d/%m/%Y")
+            )
 
-    if answer_type == "MonthYearDate":
-        return (
-            datetime.strptime(answer_value, "%Y-%m")
-            .replace(tzinfo=timezone.utc)
-            .strftime("%m/%Y")
-        )
+        if answer_type == "MonthYearDate":
+            return (
+                datetime.strptime(answer_value, "%Y-%m")
+                .replace(tzinfo=timezone.utc)
+                .strftime("%m/%Y")
+            )
 
     return answer_data
 
 
-def _get_checkbox_answer_data(answer_store, answer_schema, value):
+def _get_checkbox_answer_data(
+    answer_store: AnswerStore, answer_schema: Mapping, value: AnswerValueTypes
+) -> dict:
     qcodes_and_values = []
-    for user_answer in value:
-        # find the option in the schema which matches the users answer
-        option = next(
-            (
-                option
-                for option in answer_schema["options"]
-                if option["value"] == user_answer
-            ),
-            None,
-        )
-
-        if option:
-            if "detail_answer" in option:
-                detail_answer = answer_store.get_answer(option["detail_answer"]["id"])
+    if isinstance(value, Iterable) and value:
+        for user_answer in value:
```

Comment:
I don't think we should modify the code in this way for this function, as this is only ever called when we know it is a checkbox. Therefore, I would update the type for value to `ListAnswer` and add a type ignore to callers line. That way, code changes are not needed to please type hints as the `if` would always be true.

We need to refactor the type for answer value and set it to `Any` so we can override it. That is why we have this problem, but one for another card.

Reply:
This whole `if` conditional and nesting removed, ignore added to `value` where it's being called as the type is `ListAnswer` now, see below.